Wikipedia/UI-V5/WMFArticleContainerViewController.m

```diff
@@ -557,6 +576,71 @@ - (void)sectionEditorFinishedEditing:(SectionEditorViewController*)sectionEditor
     [self fetchArticle];
 }
 
+
+
+
+
+
+
+
+
+
+
+
+#pragma mark - UIViewControllerPreviewingDelegate
+
+- (nullable UIViewController*)previewingContext:(id<UIViewControllerPreviewing>)previewingContext
+                      viewControllerForLocation:(CGPoint)location {
+
+//return
+//[[WMFArticleContainerViewController alloc] initWithArticleTitle:[[MWKTitle alloc] initWithURL:[NSURL URLWithString:@"http://en.wikipedia.org/wiki/Cat"]]
+//                                                      dataStore:self.dataStore
+//                                                discoveryMethod:self.discoveryMethod];
+
+
+    
+    JSValue *element = [[self.webViewController.webView wmf_javascriptContext][@"getElementFromPoint"] callWithArguments:@[@(location.x), @(location.y)]];
+    
+    if([element hasProperty:@"tagName"]){
+        JSValue *v = [element valueForProperty:@"tagName"];
+        //NSLog(@"tagName = %@", [v toString]);
+        if ([[v toString] isEqualToString:@"A"]){
+            
+            /*
+             - ensure it follows links to other lang wikis!
+             - could use utilities.findClosest(event.target, 'A')
+             - exclude refs for now?
+             */
+
+            JSValue *href = [element valueForProperty:@"href"];
+NSLog(@"\n\nlink pressed: href = %@\n\n", [href toString]);
+
+            
+            
+            MWKTitle* title = [[MWKTitle alloc] initWithURL:[NSURL URLWithString:[href toString]]];
+            //NSLog(@"title = %@", title);
+
+//return nil;
+
+//BUG! The WMFArticleContainerViewController below gets dealloc'ed right away...
+            return
+            [[WMFArticleContainerViewController alloc] initWithArticleTitle:title
+                                                                  dataStore:self.dataStore
+                                                            discoveryMethod:self.discoveryMethod];
+        }
```

Comment:
My guess is that the preview is being cancelled by something. Have a close look at the stack track from w/in `dealloc` to try to find out what's triggering it